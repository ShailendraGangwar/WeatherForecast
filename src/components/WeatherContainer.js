import React, { Component } from "react";
import { View, Text, StyleSheet, Image } from "react-native";
import Icon from "react-native-vector-icons/MaterialIcons";
const WeatherContainer = props => {
  return (
    <View style={styles.container}>
      <View style={styles.headerContainer}>
        <Text style={styles.weatherHeader}>{props.currentDate}</Text>
      </View>
      <View style={styles.innerContainer}>
        <Image
          style={styles.weatherIcon}
          source={{
            uri: `https://openweathermap.org/img/w/${
              props.currentWeatherIcon
            }.png`
          }}
        />
        <View style={styles.subContainer}>
          <Text style={styles.weatherDetail}>{props.currentWeather}</Text>
          {/* <Text style={styles.weatherDetailTemp}>
            {props.currentWeatherTemp}˚C
          </Text> */}
          <View style={styles.tempContainer}>
            <Icon name="arrow-drop-down" size={30} color="red" />
            <Text style={styles.weatherDetailTempMinMax}>
              {props.currentWeatherMinTemp}˚C
            </Text>
            <Icon name="arrow-drop-up" size={30} color="red" />
            <Text style={styles.weatherDetailTempMinMax}>
              {props.currentWeatherMaxTemp}˚C
            </Text>
          </View>
        </View>
        <View style={styles.humidityContainer}>
          <Text style={styles.weatherDetailTempMinMax}>Humidity</Text>
          <Text style={styles.weatherDetailTemp}>
            {props.currentWeatherHumidity}%
          </Text>
        </View>
      </View>
    </View>
  );
};
export default WeatherContainer;

const styles = StyleSheet.create({
  container: {
    width: "100%",
    backgroundColor: "powderblue",
    flexDirection: "column",
    justifyContent: "space-evenly",
    alignItems: "center"
  },
  subContainer: {
    width: "50%",
    backgroundColor: "#93d2ed",
    flexDirection: "column",
    justifyContent: "space-evenly",
    alignItems: "flex-start"
  },
  humidityContainer: {
    //width: "50%",
    backgroundColor: "#93d2ed",
    flexDirection: "column",
    justifyContent: "space-evenly",
    alignItems: "flex-start"
  },
  tempContainer: {
    width: "70%",
    backgroundColor: "#93d2ed",
    flexDirection: "row",
    justifyContent: "flex-start",
    alignItems: "center"
  },
  innerContainer: {
    width: "100%",
    backgroundColor: "#93d2ed",
    flexDirection: "row",
    justifyContent: "space-evenly",
    alignItems: "center"
  },
  headerContainer: {
    backgroundColor: "#abdcf1",
    flexDirection: "row",
    justifyContent: "space-evenly",
    alignItems: "center"
  },
  weatherHeader: {
    width: "100%",
    fontWeight: "bold",
    margin: 2,
    textAlign: "left",
    paddingLeft: 5
  },
  weatherDetail: {
    // width: '50%',
    color: "red",
    textAlign: "center",
    margin: 2,
    paddingLeft: 5,
    fontSize: 16
  },
  weatherDetailTempMinMax: {
    color: "black",
    textAlign: "center",
    fontSize: 13,
    height: 18
  },
  weatherIcon: {
    width: 60,
    height: 60,
    resizeMode: "cover"
  },
  weatherDetailTemp: {
    color: "red",
    textAlign: "right",
    fontSize: 20,
    fontWeight: "bold",
    textAlign: "right"
  }
});
