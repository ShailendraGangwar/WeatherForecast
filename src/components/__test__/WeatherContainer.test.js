import 'react-native';
import React from 'react'
import WeatherContainer from '../WeatherContainer';

import renderer from 'react-test-renderer';

it('should renders', () => {
  const tree = renderer.create(<WeatherContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});