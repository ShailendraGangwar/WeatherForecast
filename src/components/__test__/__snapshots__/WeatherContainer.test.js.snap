// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "powderblue",
      "flexDirection": "column",
      "justifyContent": "space-evenly",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#abdcf1",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
          "margin": 2,
          "paddingLeft": 5,
          "textAlign": "left",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#93d2ed",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://openweathermap.org/img/w/undefined.png",
        }
      }
      style={
        Object {
          "height": 60,
          "resizeMode": "cover",
          "width": 60,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#93d2ed",
          "flexDirection": "column",
          "justifyContent": "space-evenly",
          "width": "70%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "red",
            "fontSize": 16,
            "margin": 2,
            "paddingLeft": 5,
            "textAlign": "center",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 13,
            "margin": 2,
            "paddingLeft": 5,
            "textAlign": "center",
          }
        }
      >
        ËšC
      </Text>
    </View>
  </View>
</View>
`;
